From f32980b436e1844a9f6045b8f432bc38cb75744c Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Wed, 20 Feb 2013 13:37:22 -0500
Subject: [PATCH] Ticket 590 - ns-slapd segfaults while trying to delete a
 tombstone entry

Bug Description:  While trying to remove a tombstone entry, ns-slapd
                  crashed with segfault.

Fix Description:  Check if op_parms->csn is NULL before dereferencing.
                  This was fixed in master via ticket 532, but this
                  patch is just part of that fix to avoid the crash.

https://fedorahosted.org/389/ticket/590

Reviewed by: Noriko(Thanks!)
(cherry picked from commit 39f19ae08afe28eb36eabe76b2add1e84c7ed805)
---
 ldap/servers/plugins/replication/repl5_plugins.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ldap/servers/plugins/replication/repl5_plugins.c b/ldap/servers/plugins/replication/repl5_plugins.c
index f7677dd..e3c3083 100644
--- a/ldap/servers/plugins/replication/repl5_plugins.c
+++ b/ldap/servers/plugins/replication/repl5_plugins.c
@@ -1064,7 +1064,7 @@ write_changelog_and_ruv (Slapi_PBlock *pb)
 			op_params->target_address.uniqueid = slapi_ch_strdup (uniqueid);
 		} 
 
-		if( is_cleaned_rid(csn_get_replicaid(op_params->csn))){
+		if( op_params->csn && is_cleaned_rid(csn_get_replicaid(op_params->csn))){
 			/* this RID has been cleaned */
 			object_release (repl_obj);
 			return 0;
-- 
1.7.7.6

